diff --git a/ts2panda/src/debuginfo.ts b/ts2panda/src/debuginfo.ts
index 6d05c0a..2fdd8a1 100644
--- a/ts2panda/src/debuginfo.ts
+++ b/ts2panda/src/debuginfo.ts
@@ -174,8 +174,25 @@ export class DebugInfo {
             if (!file) {
                 return;
             }
-            let loc = file.getLineAndCharacterOfPosition(firstStmt.getStart());
-            let wholeLineText = firstStmt.getText();
+
+            let pos : number = 0;
+            let tempWholeLineText : string = ""
+            if (firstStmt.pos === -1 || firstStmt.end === -1) {
+                let parent = firstStmt.parent;
+                while (parent) {
+                    if (parent.pos !== -1 && parent.end !== -1) {
+                        pos = parent.pos;
+                        tempWholeLineText = parent.getText();
+                        break;
+                    }
+                    parent = parent.parent;
+                }
+            } else {
+                pos = firstStmt.getStart();
+            }
+
+            let loc = file.getLineAndCharacterOfPosition(pos);            
+            let wholeLineText = tempWholeLineText || firstStmt.getText();
             posInfo.setSourecLineNum(loc.line);
             posInfo.setSourecColumnNum(loc.character);
             posInfo.setWholeLine(wholeLineText);
@@ -206,8 +223,25 @@ export class DebugInfo {
             if (!file) {
                 return;
             }
-            let loc = file.getLineAndCharacterOfPosition(tsNode.getStart());
-            wholeLineText = tsNode.getText();
+
+            let pos : number = 0;
+            let tempWholeLineText : string = ""
+            if (tsNode.pos === -1 || tsNode.end === -1) {
+                let parent = tsNode.parent;
+                while (parent) {
+                    if (parent.pos !== -1 && parent.end !== -1) {
+                        pos = parent.pos;
+                        tempWholeLineText = parent.getText();
+                        break;
+                    }
+                    parent = parent.parent;
+                }
+            } else {
+                pos = tsNode.getStart();
+            }
+
+            let loc = file.getLineAndCharacterOfPosition(pos);
+            wholeLineText = tempWholeLineText || tsNode.getText();
             lineNumber = loc.line;
             columnNumber = loc.character;
         }
